lib/transforms/addPrecacheServiceWorker.js

```diff
@@ -0,0 +1,69 @@
+var _ = require('lodash');
+var AssetGraph = require('../AssetGraph');
+var query = AssetGraph.query;
+var urlTools = require('urltools');
+var swPrecache = require('sw-precache');
+var Promise = require('bluebird');
+
+module.exports = function (queryObj) {
+    return function addPrecacheServiceWorker(assetGraph) {
+        return Promise.map(assetGraph.findAssets(_.extend({type: 'Html'}, queryObj)), function (htmlAsset) {
+            if (htmlAsset.isInline) {
+                throw new Error('addCacheManifestSinglePage: Cannot generate a CacheManifest for an inline Html asset');
+            }
+            var dynamicUrlToDependencies = {};
+            var assetByUrl = {};
+            assetGraph.eachAssetPostOrder(htmlAsset, {type: query.not(['HtmlAnchor', 'SvgAnchor', 'HtmlMetaRefresh', 'HtmlCacheManifest', 'HtmlConditionalComment', 'JavaScriptSourceUrl', 'JavaScriptSourceMappingUrl', 'JavaScriptSourceMap'])}, function (asset) {
+                // But only if the asset isn't inline, has been loaded, and isn't already in the manifest:
+                if (!asset.isInline && asset.isLoaded && asset !== htmlAsset) {
+                    var url = require('urltools').buildRootRelativeUrl(assetGraph.root, asset.url, assetGraph.root);
+                    assetByUrl[url] = asset;
+                    dynamicUrlToDependencies[url] = asset.isText ? asset.text : asset.rawSrc;
```

Comment:
It mirrors the config property accepted by sw-precache, but I guess you're right that it sounds a bit weird in this context :)